### ISSUE TITLE
SC3049 firing on when -- arg terminator used on trap

### ISSUE THREAD

**Issue by actualben** [koalaman/shellcheck#2715]
#### For bugs
- Rule Id (if any, e.g. SC1000): **SC3049**
- My shellcheck version (`shellcheck --version` or "online"): **0.9.0**
- [x] The rule's wiki page does not already cover this: **I reviewed [SC3049](https://www.shellcheck.net/wiki/SC3049)**
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

This has the error:
```sh
#!/bin/sh
trap -- 'echo "psql exited $?"' USR2
```

...taking out the `--` removes the error:
```sh
trap 'echo "psql exited $?"' USR2
```

#### Here's what shellcheck currently says:

```
[Line 3:]
trap -- 'echo "psql exited $?"' USR2
        ^-- [SC3049](https://www.shellcheck.net/wiki/SC3049) (warning): In POSIX sh, using lower/mixed case for signal names is undefined.
```

#### Here's what I wanted or expected to see:

The opengroup [manpage for trap](https://www.unix.com/man-page/POSIX/1posix/trap/) suggests that running trap without arguments prints the current signal handlers in a format "suitable for reinput to the shell as commands". That format includes the `--`; for example:

```sh
/ $ trap 'date' USR2
/ $ trap
trap -- 'date' USR2
/ $ 
```
